jtransc-rt/src/java/lang/Thread.java

```diff
@@ -163,7 +163,7 @@ private void _init() {
 	@JTranscMethodBody(target = "cpp", value = "t_ = boost::thread(&{% CLASS java.lang.Thread:runInternal %}::{% METHOD java.lang.Thread:runInternal:()V %}, this);")
 
 	public synchronized void start() {
-		JTranscThreading.impl.start(this);
+		JTranscThreading.impl.start_target(target);
```

Comment:
I think that you are doing this because target is private. In JTransc you can reference/access private fields from outside.

So here you can do two things:

One could be:

This option calls `impl.start` for all targets but Haxe/C++ where it calls start_target:

```
@HaxeMethodBody(target ="cpp", value = "{% SFIELD com.jtransc.thread.JTranscThreading:impl %}.{% METHOD com.jtransc.thread.JTranscThreading:start_target %}(this.{% FIELD #CLASS:target %});")
public synchronized void start() {
    JTranscThreading.impl.start(this);
}
```

```
SFIELD -> static field
METHOD -> instance method
FIELD -> instance field
#CLASS -> references the class of the generating method
```

Instead of figuring out the mangling, you should use it. Because the mangling cna change between calls (for example in minification) and allows me to improve the mangling in the future.

The other option is to call always start, and put the annotation in the `JTranscThreading.start` method (commented there)